click_repl/__init__.py

```diff
@@ -7,6 +9,54 @@
 import os
 import shlex
 import sys
+import six
+from .exceptions import InternalCommandException, ExitReplException
+
+__internal_commands__ = dict()
+
+
+def _register_internal_command(names, target, description=None):
+    assert hasattr(target, '__call__'), 'internal command must be a callable'
+    global __internal_commands__
```

Comment:
No, the `global` statement.

You can modify the dictionary just fine without `global`. You do need it when you want to bind a different object to that name.